src/Hosting/Abstractions/src/IHostApplicationLifetime.cs

```diff
@@ -16,14 +16,14 @@ public interface IHostApplicationLifetime
         CancellationToken ApplicationStarted { get; }
 
         /// <summary>
-        /// Triggered when the application host is performing a graceful shutdown.
-        /// Shutdown will block until this event completes.
+        /// Triggered when the application host is beginning a graceful shutdown.
+        /// Shutdown will block until any callbacks registered on this token have returned.
```

Comment:
```suggestion
        /// Shutdown will block until all callbacks registered on this token have completed.
```

I think "all" is best here. Also "completed" reads better IMO.